src/core_plugins/timelion/server/series_functions/yaxis.js

```diff
@@ -58,7 +74,13 @@ module.exports = new Chainable('yaxis', {
       myAxis.axisLabelColour = color;
       myAxis.axisLabelUseCanvas = true;
 
-
+      if (units) {
+        var unitTokens = units.split(':');
+        if (!tickFormatters[unitTokens[0]]) {
+          throw new Error (`${units} is not a supported unit type.`);
+        }
+        myAxis.units = unitTokens;
```

Comment:
For the sake of API consistency and readability of the returned object, I'd rather see this returned as 

```
{
  "type": "someType",
  "prefix": "foo",
  "suffix": "bar"
}
```